src/libwalletqt/WalletManager.cpp

```diff
@@ -86,6 +86,21 @@ Wallet *WalletManager::recoveryWallet(const QString &path, const QString &memo,
     return m_currentWallet;
 }
 
+Wallet *WalletManager::createWalletFromKeys(const QString &path, const QString &language, bool testnet,
+                                            const QString &address, const QString &viewkey, const QString &spendkey,
+                                            quint64 restoreHeight)
+{
+    QMutexLocker locker(&m_mutex);
+    if (m_currentWallet) {
+        qDebug() << "Closing open m_currentWallet" << m_currentWallet;
+        delete m_currentWallet;
```

Comment:
I meant the pointer is not function local, but class local, and will be pointing to a deleted object, so I'd set it to NULL if an exception can be thrown before it is reset in this function.